react/ProductListWrapper.tsx

```diff
@@ -18,6 +18,19 @@ interface Props {
 const ProductListWrapper: StorefrontFunctionComponent<Props> = props => {
   const { loading } = useOrderForm()
   const { device } = useDevice()
+  const hasMobileContent = Boolean(
+    useChildBlock({ id: 'product-list-content-mobile' })
+  )
+  const hasDesktopContent = Boolean(
+    useChildBlock({ id: 'product-list-content-desktop' })
+  )
+
+  if (!hasMobileContent) {
+    return <ExtensionPoint id="product-list-content-desktop" {...props} />
+  }
+  if (!hasDesktopContent) {
+    return <ExtensionPoint id="product-list-content-mobile" {...props} />
+  }
```

Comment:
Maybe?

```suggestion
if (hasMobileContent) {
    return <ExtensionPoint id="product-list-content-mobile" {...props} />
  }
  if (hasDesktopContent) {
    return <ExtensionPoint id="product-list-content-desktop" {...props} />
  }
```

Reply:
I see what you mean! But that's not quite what I'm trying to accomplish.
The behavior I'm going for is the following:
- If only `product-list-content-desktop` is passed to `product-list`, then it should always be rendered;
- If only `product-list-content-mobile` is passed to `product-list`, then it should always be rendered;
- If *both* are passed, then they should be rendered according to the screen size. That's controlled:
```tsx
return (
    <SizeMe noPlaceholder>
      {({ size }) =>
        (loading && device === 'phone') ||
        (size.width && size.width < MAX_MOBILE_WIDTH) ? (
          <ExtensionPoint id="product-list-content-mobile" {...props} />
        ) : (
          <ExtensionPoint id="product-list-content-desktop" {...props} />
        )
      }
    </SizeMe>
  )
```